addons/website/static/src/snippets/s_embed_code/options.js

```diff
@@ -0,0 +1,86 @@
+odoo.define('website.s_embed_code_options', function (require) {
+'use strict';
+
+const Dialog = require('web.Dialog');
+const core = require('web.core');
+const options = require('web_editor.snippets.options');
+
+const _t = core._t;
+
+options.registry.EmbedCode = options.Class.extend({
+    events: {
+        'click .o_we_edit_code': '_onEditClick',
+    },
+
+    _onEditClick: async function () {
```

Comment:
Use ES6 syntax for internal function definition:

`async myFunction() {`

But see, here your method is an async handler... but the async part is basically useless and unawaited by the editor system. Declared as an option method, that async part would fit in the system and ensure for exemple that a ctrl-z after your edition will undo this change... and not a potential change that happened meanwhile.